src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -938,18 +923,15 @@ types .* or update switch properties and add needed encapsulation type./).matche
 
         when: "Update the flow: switch id on the dst endpoint"
         def newDstSwitch = allSwitches[2]
-        flowHelperV2.updateFlow(flow.flowId, flow.tap {
-            it.destination.switchId = newDstSwitch.dpId
-            it.destination.portNumber = updatedFlow.source.portNumber
-        })
+        updateRequest.destination.switchId = newDstSwitch.dpId
+
+        updatedFlow = updatedFlow.update(updateRequest).waitForBeingInState(UP)
```

Comment:
What if `update()` would accept FlowExtended instead of DTO? You'd be able to hide `.toUpdateRequest()` call inside FlowExtended class and the test would look like:
```
 def flow = flowFactory.getRandom(switchPair)
def updatedFlow = flow.setDestinationSwitchId(newSwitchId)
verifyAll(flow.updateTo(updatedFlow) {
    it.destination.switchId = updatedFlow.destination.switchId
    ...
}
```
Not sure that it's more readable, though, but at least we don't mentions northbound calls (FlowUpdateDto) in tests.

Reply:
updated and added method hasTheSamePropertiesAs() to check actual and expected flow entity